src/main/java/seedu/duke/InternshipList.java

```diff
@@ -217,38 +228,84 @@ public void listInternshipsSortedByDeadline() {
         // Sort by the first skill alphabetically
         Collections.sort(sortedList, Comparator.comparing(internship -> {
             String firstSkill = internship.getFirstSkill();
             return firstSkill.isEmpty() ? "No skills" : firstSkill.toLowerCase();// Sort internships with no skills last
         }));
-
-        // Display the sorted list without changing IDs
-        ui.showInternships(sortedList, "skills");
     }
 
     /**
      * Lists internships sorted by status alphabetically (case-insensitive).
      */
     public void listInternshipsSortedByStatus() {
-        ArrayList<Internship> sortedList = new ArrayList<>(internships);
+        ArrayList<Internship> sortedInternships = new ArrayList<>(internships);
 
-        // Sort by status alphabetically
-        Collections.sort(sortedList, Comparator.comparing(internship -> internship.getStatus().toLowerCase()));
+        Collections.sort(sortedInternships, Comparator.comparing(internship -> internship.getStatus().toLowerCase()));
+        ui.showInternships(sortedInternships, "status");
+    }
+
+    /**
+     * List all favourite internships in sorted order by role alphabetically (case-insensitive)
+     */
+    public void listFavouriteInternshipsSortedByRole() {
+        ArrayList<Internship> sortedInternships = new ArrayList<>(favouriteInternships);
+        Collections.sort(sortedInternships, Comparator.comparing(internship -> internship.getRole().toLowerCase()));
+        ui.showInternships(sortedInternships, "role in favourite");
+    }
+
+    /**
+     * List all internships sorted by duration (year first), then end date
+     */
+    public void listFavouriteInternshipsSortedByDuration() {
+        ArrayList<Internship> sortedInternships = new ArrayList<>(favouriteInternships);
+        sortedByDurationFunction(sortedInternships);
+        ui.showInternships(sortedInternships, "duration in favourite");
+    }
+
+    /**
+     * List all favourite internships sorted by the earliest deadline
+     */
+    public void listFavouriteInternshipsSortedByDeadline() {
+        ArrayList<Internship> sortedInternships = new ArrayList<>(favouriteInternships);
+
+        sortedByDeadlineFunction(sortedInternships);
+        ui.showInternships(sortedInternships, "deadline in favourite");
+    }
+
+    /**
+     * Lists all favourite internships sorted by the first skill alphabetically (case-insensitive).
+     */
+    public void listFavouriteInternshipsSortedByFirstSkill() {
+        ArrayList<Internship> sortedInternships= new ArrayList<>(favouriteInternships);
 
-        // Display the sorted list without changing IDs
-        ui.showInternships(sortedList, "status");
+        sortedByFirstSkillFunction(sortedInternships);
+        ui.showInternships(sortedInternships, "skills in favourite");
     }
 
+    /**
+     * Lists all favourite internships sorted by status alphabetically (case-insensitive).
+     */
+    public void listFavouriteInternshipsSortedByStatus() {
```

Comment:
these functions seem like they are double work, as the code written to sort the favourites list is exactly the same as sorting the entire list. is there a way to simplify the code?